src/filelock/_soft.py

```diff
@@ -10,8 +14,11 @@ def _acquire(self):
         open_mode = os.O_WRONLY | os.O_CREAT | os.O_EXCL | os.O_TRUNC
         try:
             fd = os.open(self._lock_file, open_mode)
-        except OSError:
-            pass
+        except OSError as exception:
+            if exception.errno in (EPERM, EACCES, ENOENT):
+                raise
+            if sys.platform == "win32" and not os.access(self._lock_file, os.W_OK):
+                raise PermissionError("Permission denied")
```

Comment:
Maybe `raise ... from None` to remove the exception chaining.

Reply:
that doesn't work on python 2, that we still support here

Reply:
Ah, of course.